### DISCUSSION TITLE
Get length of paginated list

### DISCUSSION THREAD

**Discussion by datavoyager** [ucfopen/canvasapi#586] [Q&A] [answered]
Hi Folks,

Have just come across canvasapi and am mightily impressed. 

I've trying to get the length of a paginated list of sections:

```
    sections = course.get_sections()
    if len(sections) == 0:
        raise Exception("No sections exist in {}".format(course.sis_course_id))
```

len() doesn't appear to work. Is there any way to get the paginated list size or is it not possible due to the way they are retrieved?

Best wishes,
Luke

**Comment by bennettscience** [comment, marked as answer]
To get the length, it needs to be cast into a `list`. So, `len(list(sections))` will give you the length quickly.

**Reply by Thetwam** [reply]
Note that calling `list` on a `PaginatedList` will evaluate the entire thing. This means that (depending on which endpoint you're looking at) it could result in many extra API calls. In the case of Sections there probably are rarely (if ever) more than 100 sections in a course so it's probably safe, but keep that in mind for other endpoints.

**Reply by datavoyager** [reply]
Thanks both - your replies are greatly appreciated.
